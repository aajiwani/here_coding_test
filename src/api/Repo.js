import APICaller from "lib/ApiCaller";
import _ from "lodash";

export function GetLanguages(repoName) {
  var apiUrl = "https://api.github.com/repos/heremaps/@name/languages".replace(
    "@name",
    repoName
  );

  return APICaller.Call(apiUrl).then(response => {
    return _.map(Object.keys(response), key => {
      return {
        name: key,
        bytes: response[key]
      };
    });
  });
}
