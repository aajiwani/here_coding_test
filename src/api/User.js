import APICaller from "lib/ApiCaller";
import _ from "lodash";

export function GetRepositories() {
  var apiUrl = "https://api.github.com/users/heremaps/repos";

  return APICaller.Call(apiUrl).then(response => {
    return _.map(response, val => {
      return {
        name: val.name,
        full_name: val.full_name,
        image: val.owner.avatar_url
      };
    });
  });
}
