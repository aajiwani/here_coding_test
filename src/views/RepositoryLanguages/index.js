import React from "react";
import ReactPromisedComponent from "react-promised-component";
import LoadingComponent from "components/Loading";
import ErrorComponent from "components/Error";
import SuccessComponent from "./components/success_component.js";
import * as RepoApi from "api/Repo";

var PromisedReactComponent = ReactPromisedComponent(
  "promise_name",
  LoadingComponent,
  ErrorComponent,
  SuccessComponent
);

export default class PromisedComponent extends React.Component {
  constructor(props) {
    super(props);
  }

  // Promise creator method
  promiseGenerator(params) {
    return RepoApi.GetLanguages(this.props.match.params.repo);
  }

  render() {
    return (
      <PromisedReactComponent
        promise_name={this.promiseGenerator.bind(this)}
        repo_name={this.props.match.params.repo}
        goBack={() => this.props.history.goBack()}
      />
    );
  }
}
