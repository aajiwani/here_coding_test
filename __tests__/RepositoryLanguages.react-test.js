jest.mock('lib/ApiCaller');

import React from 'react';
import RepositoryLanguages from 'views/RepositoryLanguages/components/success_component';
import renderer from 'react-test-renderer';
import {MemoryRouter, Switch, Route} from 'react-router-dom';
import * as RepoApi from 'api/Repo';

test('Renders the list of repositories owned by heremaps', () => {
  const repoName = 'bike-navigation';
  return RepoApi.GetLanguages(repoName).then(output => {
    const component = renderer.create(
      <MemoryRouter
        initialEntries={['/repositories', '/languages/:repo']}
        initialIndex={0}
      >
        <RepositoryLanguages
          result={output}
          repo_name={repoName}
          goBack={() => {}}
        />
      </MemoryRouter>
    );

    expect(output).toBeDefined();

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
