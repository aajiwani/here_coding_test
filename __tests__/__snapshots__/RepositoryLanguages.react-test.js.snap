// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the list of repositories owned by heremaps 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className=""
      style={
        Object {
          "display": "flex",
          "justifyContent": "flex-start",
        }
      }
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="text-center col-md-6 col-xs-12 col-xs-offset-6"
    >
      <h2
        style={
          Object {
            "color": "grey",
          }
        }
      >
        bike-navigation
      </h2>
    </div>
  </div>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          Language
        </th>
        <th>
          # of Bytes
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          Eagle
        </td>
        <td>
          1335474
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          C
        </td>
        <td>
          12754
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          C++
        </td>
        <td>
          5356
        </td>
      </tr>
      <tr>
        <td>
          4
        </td>
        <td>
          Arduino
        </td>
        <td>
          5317
        </td>
      </tr>
      <tr>
        <td>
          5
        </td>
        <td>
          OpenSCAD
        </td>
        <td>
          3576
        </td>
      </tr>
      <tr>
        <td>
          6
        </td>
        <td>
          Python
        </td>
        <td>
          2322
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
